diff -rua terminal_markdown_viewer-1.6.3.b/mdv/markdownviewer.py terminal_markdown_viewer-1.6.3/mdv/markdownviewer.py
--- terminal_markdown_viewer-1.6.3.b/mdv/markdownviewer.py	2016-12-08 04:26:24.000000000 +0100
+++ terminal_markdown_viewer-1.6.3/mdv/markdownviewer.py	2018-07-30 16:51:19.408298758 +0200
@@ -126,9 +126,11 @@
 """
 
 
+import sys
+# grrr
+PY3 = True if sys.version_info[0] > 2 else False
 import io
 import os
-import sys
 import textwrap
 is_app = 0
 # code analysis for hilite:
@@ -158,6 +160,12 @@
     from HTMLParser import HTMLParser
 envget = os.environ.get
 
+
+# adaptions:
+if PY3:
+    unichr = chr
+
+
 # ---------------------------------------------------------------------- Config
 hr_sep, txt_block_cut, code_pref, list_pref, bquote_pref, hr_ends = \
     '─', '✂', '| ', '- ', '|', '◈'
