diff -u -r ../src.old/Crypto/AesSmall_x86.asm Crypto/AesSmall_x86.asm
--- Crypto/AesSmall_x86.asm	2016-02-14 10:29:58.000000000 +0100
+++ Crypto/AesSmall_x86.asm	2016-08-05 13:17:06.828744123 +0200
@@ -1442,3 +1442,21 @@
     db  v8(0xe1),v8(0x69),v8(0x14),v8(0x63),v8(0x55),v8(0x21),v8(0x0c),v8(0x7d)
 
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
+
+%ifidn __OUTPUT_FORMAT__,elf32
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
+
+%ifidn __OUTPUT_FORMAT__,elf64
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
Only in Crypto: AesSmall_x86.asm.old
diff -u -r ../src.old/Crypto/Aes_hw_cpu.asm Crypto/Aes_hw_cpu.asm
--- Crypto/Aes_hw_cpu.asm	2016-02-14 10:29:58.000000000 +0100
+++ Crypto/Aes_hw_cpu.asm	2016-08-05 13:14:38.025523108 +0200
@@ -328,3 +328,13 @@
 
 
 %endif	; __BITS__ != 16
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
Only in Crypto: Aes_hw_cpu.asm.old
diff -u -r ../src.old/Crypto/Aes_x64.asm Crypto/Aes_x64.asm
--- Crypto/Aes_x64.asm	2016-02-14 10:29:58.000000000 +0100
+++ Crypto/Aes_x64.asm	2016-08-05 13:18:05.420012831 +0200
@@ -905,3 +905,21 @@
 %endif
 
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
+
+%ifidn __OUTPUT_FORMAT__,elf32
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
+
+%ifidn __OUTPUT_FORMAT__,elf64
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
Only in Crypto: Aes_x64.asm.old
diff -u -r ../src.old/Crypto/Aes_x86.asm Crypto/Aes_x86.asm
--- Crypto/Aes_x86.asm	2016-02-14 10:29:58.000000000 +0100
+++ Crypto/Aes_x86.asm	2016-08-05 13:18:41.470793468 +0200
@@ -644,3 +644,21 @@
     do_exit
 
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
+
+%ifidn __OUTPUT_FORMAT__,elf32
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
+
+%ifidn __OUTPUT_FORMAT__,elf64
+
+    section .note.GNU-stack noalloc noexec nowrite progbits
+
+%endif
Only in Crypto: Aes_x86.asm.old
